compiler/backend/src/org/jetbrains/kotlin/codegen/coroutines/CoroutineTransformerMethodVisitor.kt

```diff
@@ -657,7 +657,9 @@ class CoroutineTransformerMethodVisitor(
             // Also, we spill and unspill null for visible dead variables
             // `generateSpillAndUnspill` calls the probe from stdlib for us.
             for (referenceToSpill in referencesToSpillBySuspensionPointIndex[suspensionPointIndex]) {
-                generateSpillAndUnspill(methodNode, suspension, referenceToSpill, suspendLambdaParameters)
+                // skip restoring this
+                val isInstanceThisVariable = !isStatic(methodNode.access) && referenceToSpill.slot == 0
```

Comment:
Please, move the variable into the `generateSpillAndUnspill` function. This way, you get rid of the parameter.

Reply:
> Please, move the variable into the `generateSpillAndUnspill` function. This way, you get rid of the parameter.

done

Reply:
done